gapis/api/vulkan/frame_loop.go

```diff
@@ -781,29 +781,50 @@ func (f *frameLoop) detectChangedResources(ctx context.Context) {
 
 func (f *frameLoop) detectChangedBuffers(ctx context.Context) {
 
-	apiState := GetState(f.loopEndState)
+	apiState := GetState(f.loopStartState)
 
 	// Find out changed buffers.
 	for bufferKey, buffer := range apiState.Buffers().All() {
 
-		data := buffer.Memory().Data()
-		span := interval.U64Span{data.Base(), data.Base() + data.Size()}
-		poolID := data.Pool()
+		toDestroy := f.bufferToDestroy[buffer.VulkanHandle()]
+		toCreate := f.bufferToCreate[buffer.VulkanHandle()]
+
+		if toCreate == true {
+
+			// If we're going to recreate this object for the start of the loop we need to set its state back to the right conditions
+			f.bufferChanged[bufferKey] = true
+			continue
+
+		} else if toDestroy == true {
+
+			// If we created this object during the loop and we're going to destroy this object at the end of the loop then we don't need to capture the state
+			continue
+
+		} else {
+
+			// Otherwise, we'll need to capture this objects state IFF it was modified during the loop.
+
+			data := buffer.Memory().Data()
+			span := interval.U64Span{data.Base(), data.Base() + data.Size()}
+			poolID := data.Pool()
+
+			// Did we see this buffer get written to during the loop? If we did, then we need to capture the values at the start of the loop.
+			// TODO: This code does not handle the possibility of new DeviceMemory being bound to the object during the loop. Talk to @purvisa.
```

Comment:
Prefer: `TODO(@purvisa): ...`
Also, don't hesitate to create a bug referring to this, and to add the bugID, e.g. `TODO(42123): ...`, to make sure it is actually tracked.